IdentityCore/src/webview/operations/MSIDSwitchBrowserResumeOperation.m

```diff
@@ -0,0 +1,101 @@
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+
+#import "MSIDSwitchBrowserResumeOperation.h"
+#import "MSIDSwitchBrowserResumeResponse.h"
+#import "MSIDWebviewFactory.h"
+#import "MSIDInteractiveTokenRequestParameters.h"
+#import "MSIDWebResponseOperationFactory.h"
+
+@interface MSIDSwitchBrowserResumeOperation()
+
+@property (nonatomic) MSIDSwitchBrowserResumeResponse *switchBrowserResumeResponse;
+
+@end
+
+
+@implementation MSIDSwitchBrowserResumeOperation
+
++ (void)load
+{
+    [MSIDWebResponseOperationFactory registerOperationClass:self forResponseClass:MSIDSwitchBrowserResumeResponse.class];
+}
+
+- (nullable instancetype)initWithResponse:(MSIDWebviewResponse *)response
+                                    error:(NSError *__autoreleasing *)error
+{
+    self = [super initWithResponse:response error:error];
+    if (self)
+    {
+        if (![response isKindOfClass:MSIDSwitchBrowserResumeResponse.class])
+        {
+            NSString *errorMsg = [NSString stringWithFormat:@"%@ is required for creating %@", MSIDSwitchBrowserResumeResponse.class, self.class];
+            MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"%@", errorMsg);
+            if (error)
+            {
+                *error = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, errorMsg, nil, nil, nil, nil, nil, NO);
+            }
+            
+            return nil;
+        }
+        
+        _switchBrowserResumeResponse = (MSIDSwitchBrowserResumeResponse *)response;
+    }
+    
+    return self;
+}
+
+- (void)invokeWithRequestParameters:(nonnull MSIDInteractiveTokenRequestParameters *)requestParameters
+            webRequestConfiguration:(MSIDAuthorizeWebRequestConfiguration *)webRequestConfiguration
+                       oauthFactory:(nonnull MSIDOauth2Factory *)oauthFactory
+  decidePolicyForBrowserActionBlock:(nullable MSIDExternalDecidePolicyForBrowserActionBlock)decidePolicyForBrowserActionBlock
+     webviewResponseCompletionBlock:(nonnull MSIDWebviewAuthCompletionHandler)webviewResponseCompletionBlock
+   authorizationCodeCompletionBlock:(nonnull MSIDInteractiveAuthorizationCodeCompletionBlock)authorizationCodeCompletionBlock
+{
+    webRequestConfiguration.startURL = [[NSURL alloc] initWithString:self.switchBrowserResumeResponse.actionUri];
```

Comment:
Check that actionUri is not null here? I think NSURL init will crash if you pass it a null uri

Reply:
It is marked as "nonnull" and we verify it in the initial of MSIDSwitchBrowserResponse (it is parent of MSIDSwitchBrowserResumeResponse). If it is nil or blank -- we won't even create MSIDSwitchBrowserResumeResponse.